### PR TITLE
[UOE] Update events with a correct flow parameter

### PR DESCRIPTION
**Pull request by wzieba**
Relates to: #6711 

### Description

This PR adds a parameter of `flow: editting` for cases when VM actions are executed in context of editing. 

The `*_add` events are also triggered in case of edit. It's in sync with iOS and has been discussed here: p1657631639043799-slack-C02KUCFCSFP

### Testing instructions

As the code is covered with unit tests, I can only recommend a look at the logs (tag: `WooCommerce-UTILS`) when executing actions:
- product select
- customer note edit
- status change
- shipping edit
- fee edit

in both create and edit contexts. 

### Images/gif

None

- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt [hunk 1]
```diff
@@ -223,6 +223,12 @@ class AnalyticsTracker private constructor(private val context: Context) {
         const val ORDER_EDIT_SHIPPING_ADDRESS = "shipping_address"
         const val ORDER_EDIT_BILLING_ADDRESS = "billing_address"
 
+        enum class OrderNoteType(val value: String) {
+            CUSTOMER("customer"),
+            PRIVATE("private"),
+            SYSTEM("system")
+        }
```
Comment (wzieba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/analytics/AnalyticsTracker.kt:230]: Based on `Woo Mobile Events` spreadsheet, we need to also add `type`. Discussed here: p1657631639043799-slack-C02KUCFCSFP

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/creation/UnifiedOrderEditViewModelTest.kt [hunk 2]
```diff
@@ -86,6 +94,81 @@ abstract class UnifiedOrderEditViewModelTest : BaseUnitTest() {
         determineMultipleLinesContext = mock {
             on { invoke(any()) } doReturn OrderCreationViewModel.MultipleLinesContext.None
         }
+        tracker = mock()
+    }
+
+    protected abstract val tracksFlow: String
+
+    @Test
+    fun `when product selected, send tracks event`() {
```
Comment (wzieba) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/creation/UnifiedOrderEditViewModelTest.kt:103]: I understand that those tests might feel a little dummy. Still, on the other hand, analytics are crucial in our decision-making process, and it's important to not modify or remove them when working with the codebase later. That's why I've added those tests (and I think we generally should start covering analytics behavior with unit tests).

Comment (atorresveiga) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/orders/creation/UnifiedOrderEditViewModelTest.kt:103]: I love this idea @wzieba 👏

**Review by atorresveiga — APPROVED** [review]
Great work @wzieba. I loved the work you do on unit testing analytics 🙌. I have tested the changes and everything is working as expected:

#### Create flow evidence:

Screen Shot 2022-07-14 at 09 58 48

#### Edit flow evidence:

Screen Shot 2022-07-14 at 09 46 13

:shipit:

**Review by atorresveiga — APPROVED** [review]
LGTM!

**Comment by oguzkocer** [comment]
It looks like this PR was shipped with `9.7`, so I am updating the milestone to reflect it.

**Pull request merged**